### PR TITLE
ebuild  0.14

### PR DESCRIPTION
**Pull request by happysalada**
Just adds an override for esbuild 0.14.
I'll add more of those if that works for you.

### CODE REVIEW

File: overrides/nodejs/default.nix [hunk 1]
```diff
@@ -370,6 +370,24 @@ in
           };
         in "${esbuild}/bin/esbuild";
       };
+      "add-binary-0.14" = {
+        _condition = pkg: satisfiesSemver "~0.14" pkg;
```
Comment (DavHau) [inline overrides/nodejs/default.nix:374]:
```suggestion
        _condition = satisfiesSemver "~0.14";
```

**Comment by happysalada** [comment]
@DavHau let me know if that is what you had in mind.
I'll be adding more as I go along.

**Review by DavHau — COMMENTED** [review]
currying is king

**Comment by happysalada** [comment]
haha, will remember that one.
All good now.

**Comment by DavHau** [comment]
Wonderful. Thank you very much. This solution looks good. I think we can solve similar problems like that as well.
At some point we should probably start using some input management for these extra sources. Otherwise there is a risk of those being forgotten and never updated.

**Comment by happysalada** [comment]
I completely agree.
I don't know if you've seen the patttern in nixpkgs that uses a pin.json
ideally we should have the same, a json file that stores data like version, sha256 and the like.
So that a script can update those very easily.

**Comment by DavHau** [comment]
I'm a fan of using a flake for managing inputs. We could have a separate flake.nix just for the overrides.
But then again flakes doesn't support custom update logic.
But we have already custom update logic implemented in `./src/fetchers` and `./src/updaters`. Maybe we can glue that together with with flakes somehow.
Instead of `nix flake update` you would run `nix run .#update` which is our custom updater implementation.

**Comment by happysalada** [comment]
that sounds like a good idea.
I haven't seen it where you can update to the next version for example. This sounds like it would exist though.

**Pull request merged**